ADBDEV-4618 Free tuple after use

dnskvlnk opened the pull request:
refresh_rejectmap looks for a tuple using SearchSysCacheCopy1 which retrieves
a copy of the tuple allocating memory for it. However, refresh_rejectmap didn't
free these tuple copies after use. If lots of oids were passed, diskquota could
work incorrectly because of huge memory leak. This patch frees these tuples and
prevents memory leaks.

RekGRpth commented on src/quotamodel.c:
```diff
@@ -2118,6 +2118,11 @@ refresh_rejectmap(PG_FUNCTION_ARGS)
+
+		if (HeapTupleIsValid(tuple))
+		{
+			heap_freetuple(tuple);
```
may be add same for
```c
HeapTuple curr_tuple = SearchSysCacheCopy1(RELOID, ObjectIdGetDatum(curr_oid));
```
?

dnskvlnk replied:
Done

RekGRpth commented:
> prevets

prevents

Commit:
Free cache tuple after use

refresh_rejectmap looked for an oid in local cache and retrieved a copy
in a case of success. However, such a tuple should be freed after use,
but refresh_rejectmap didn't do that which resulted in huge memory leak
if there were passed lots of oid, which could result in diskquota
stopping working. This patch frees these tuples and prevents memory leaks

bimboterminator1 commented:
@dnskvlnk , I'm ready to approve. Looks good. 
But before merging there is a new requirement to align PR description by 80 symbols as well. So, please.

andr-sokolov commented on src/quotamodel.c:
```diff
@@ -2118,6 +2120,10 @@ refresh_rejectmap(PG_FUNCTION_ARGS)
 			}
 			LWLockRelease(diskquota_locks.relation_cache_lock);
 		}
+		if (HeapTupleIsValid(tuple))
```
Why the second `if (HeapTupleIsValid(tuple))` has been added? Can we call heap_freetuple for tuple at the end of the positive branch of the first `if (HeapTupleIsValid(tuple))`?

Commit:
Call heap_freetuple at the end of the positive branch of HeapTupleIsValid
